sormas-backend/src/main/java/de/symeda/sormas/backend/person/PersonFacadeEjb.java

```diff
@@ -1859,6 +1866,84 @@ public void mergePerson(String leadPersonUuid, String otherPersonUuid, boolean m
 		service.ensurePersisted(leadPerson);
 	}
 
+	private void processPersonAddressMerge(PersonDto leadPersonDto, PersonDto otherPersonDto) {
+		if (differentLocation(leadPersonDto.getAddress(), otherPersonDto.getAddress())) {
+			LocationDto otherAddress = otherPersonDto.getAddress();
+			otherAddress.setAddressType(PersonAddressType.OTHER_ADDRESS);
+			otherAddress.setAddressTypeDetails(I18nProperties.getString(Strings.messagePersonMergedAddressDescription));
+			leadPersonDto.addAddress(otherAddress);
+		} else {
+			DtoHelper.copyDtoValues(leadPersonDto.getAddress(), otherPersonDto.getAddress(), false);
+		}
+	}
+
+	private boolean differentLocation(LocationDto firstAddress, LocationDto secondAddress) {
+
+		java.util.function.Predicate<Function<LocationDto, HasUuid>> differentInfrastructurePredicate = (Function<LocationDto, HasUuid> function) -> {
+			return function.apply(firstAddress) != null
+				&& function.apply(secondAddress) != null
+				&& !DataHelper.equal(function.apply(firstAddress).getUuid(), function.apply(secondAddress).getUuid());
+
+		};
+
+		if (Boolean.TRUE.equals(differentInfrastructurePredicate.test(LocationDto::getCountry))) {
+			return true;
+		}
+
+		if (Boolean.TRUE.equals(differentInfrastructurePredicate.test(LocationDto::getRegion))) {
+			return true;
+		}
+
+		if (Boolean.TRUE.equals(differentInfrastructurePredicate.test(LocationDto::getDistrict))) {
+			return true;
+		}
+
+		if (Boolean.TRUE.equals(differentInfrastructurePredicate.test(LocationDto::getCommunity))) {
+			return true;
+		}
+
+		if (firstAddress.getFacility() != null) {
+			FacilityDto firstAddressFacilityDto = facilityFacade.getByUuid(firstAddress.getFacility().getUuid());
+			if (secondAddress.getCommunity() != null
+				&& !firstAddressFacilityDto.getCommunity().getUuid().equals(secondAddress.getCommunity().getUuid())) {
+				return true;
+			}
+		}
+
+		boolean oneMatch = Boolean.FALSE;
+		boolean notNullValues = Boolean.FALSE;
+		boolean different = Boolean.FALSE;
+		List<Pair<String, String>> addressValues = new ArrayList<>();
+		addressValues.add(new Pair<>(firstAddress.getCity(), secondAddress.getCity()));
+		addressValues.add(new Pair<>(firstAddress.getPostalCode(), secondAddress.getPostalCode()));
+		addressValues.add(new Pair<>(firstAddress.getStreet(), secondAddress.getStreet()));
+		addressValues.add(new Pair<>(firstAddress.getHouseNumber(), secondAddress.getHouseNumber()));
+
+		for (Pair<String, String> addressTypePair : addressValues) {
+			if (addressTypePair.getElement0() != null && addressTypePair.getElement1() != null) {
```

Comment:
Use `isNotBlank` instead of a null check so that we consider null and a String of whitespaces to be equal